kolibri/auth/filters.py

```diff
@@ -0,0 +1,185 @@
+from django.db import models
+from django.db.models.query import F
+from six import string_types
+
+from .constants import collection_kinds
+
+
+class HierarchyRelationsFilter(object):
```

Comment:
This is a complex class that seems to deal directly with SQL statements. I think the implication is that we might create something that is not transferable to other database backends outside of sqlite -- is that true? It might not matter if we don't intend for Kolibri to actually be used with other database backends. But if we *do* then we should try to answer this question definitively, by setting up a test case that uses the target database backend. What do you think?

Secondly, I think this class in particular will benefit from comprehensive test coverage, but on this branch the test coverage for this class is not comprehensive, and indeed in `models.py` some usages of this class are missed. The coverage is also incomplete on #51. As a tactical review choice, I would substitute 100% test coverage for line-by-line scrutiny. Can you get 100% coverage in #51? Otherwise I would feel obliged to spend more time on this PR.